subsys/net/lib/download_client/src/download_client.c

```diff
@@ -72,7 +72,7 @@ static int socket_timeout_set(int fd)
 	return 0;
 }
 
-static int socket_sectag_set(int fd, int sec_tag)
+static int socket_sectag_set(int fd, int sec_tag, const char * const hostname)
```

Comment:
Just a suggestion, but the name of the function could be changed to something more generic because it's already setting three different TLS socket options, not only sec tag.